File: sed_repos/java/SED7/src/main/java/ic/doc/ForecasterAdapter.java
package ic.doc;

import com.weather.Forecaster;

public class ForecasterAdapter implements ForecastGenerator {

  private final Forecaster forecaster;

  public ForecasterAdapter(Forecaster forecaster) {
    this.forecaster = forecaster;
  }

  @Override
  public WeatherForecast forecastFor(WeatherRegion region, WeatherDay day) {
    return new ForecastAdapter(forecaster.forecastFor(region.getRegion(), day.getDay()));
  }
}

File: sed_repos/python/SED7/src/forecaster_adapter.py
from src.forecast_generator import ForecastGenerator
from src.forecast_adapter import ForecastAdapter


class ForecasterAdapter(ForecastGenerator):
    def __init__(self, forecaster):
        self.forecaster = forecaster

    def forecast_for(self, region, day):
        forecast = self.forecaster.forecast_for(region.get_region(),
                                                day.get_day())
        return ForecastAdapter(forecast)
